src/amber/pipes/client_ip.cr

```diff
@@ -0,0 +1,34 @@
+module Amber
+  module Pipe
+    # Handler to determine real client IP address when Amber is behind a
+    # reverse proxy and save it to context.client_ip.
+    #
+    # The proxy should pass client IP in a header such as "X-Forwarded-For".
+    # In case of multiple values in a header, the first one is used.
+    #
+    # Crystal does not currently make remote IP available, so it is impossible
+    # to accept the header value only when coming from trusted proxy IPs.
+    # For that and other benefits, it is suggested to always run Amber apps
+    # behind proxies such as HAProxy or Nginx.
```

Comment:
One nitpick here. In the first sentence you label these reverse proxies and in the last sentence just proxies. Unless I'm missing something, those should both be the same.